hash/jump/jump_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package jump
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestHashDistribution tests that Hash is distributing keys evenly among buckets
+func TestHashDistribution(t *testing.T) {
+	// 95th percentile of Chi-Squared Distribution with 9, 99, 999, 9999 degrees of freedom
+	chiSquared95Percentiles := []float64{16.9190, 123.2252, 1073.6427, 10232.7373}
+
+	numBuckets := []int64{10, 100, 1000, 10000}
+	for i, n := range numBuckets {
+		maxNum := uint64(n) * 1000
+		buckets := make(map[int64]int, n)
+
+		var hash uint64
+		for ; hash < maxNum; hash++ {
+			idx := Hash(hash, n)
+			buckets[idx]++
+		}
+
+		expected := float64(maxNum) / float64(n)
+		var testStatistic float64
+		for _, observed := range buckets {
+			diff := float64(observed) - expected
+			testStatistic += diff * diff / expected
+		}
+
+		// To test that Hash is evenly distributing keys among the buckets we'll perform Pearson's
+		// chi-squared test with a significance level of 95%.
+		assert.True(t, testStatistic < chiSquared95Percentiles[i])
+	}
+}
+
+// TestHashMoved tests that Hash only redistributes approximately 1/(n+1) keys when going from
+// n to n+1 buckets
+func TestHashMoved(t *testing.T) {
+	numBuckets := []int64{10, 100, 1000}
+
+	for _, n := range numBuckets {
+		maxNum := uint64(n) * 1000
```

Comment:
Instead of choosing `[0, n*1000)` to test I would instead check `[0, math.MaxUint64)` with a step size of `math.MaxUint64 / (n * 1000)`.  This will test the entire keyspace instead of a subset.